crates/ruff_python_index/src/indexer.rs

```diff
@@ -228,8 +228,18 @@ impl Indexer {
         offset: TextSize,
         locator: &Locator,
     ) -> Option<TextSize> {
-        // Find the first preceding continuation.
-        let mut continuation = self.find_continuation(locator.line_start(offset), locator)?;
+        // Find the first preceding continuation. If the offset isn't the first non-whitespace
+        // character on the line, then we can't have a continuation.
+        let previous_line_end = locator.line_start(offset);
+        if !locator
+            .slice(TextRange::new(previous_line_end, offset))
+            .chars()
+            .all(is_python_whitespace)
+        {
+            return None;
```

Comment:
Nit: this is not the end of the previous line (its the full end because it includes the newline) but the start of the current lint

Should we create a helper for this on Locator. I think it's the second time in 7 days that you wrote this exact code